hypervisor/hw/pci.c

```diff
@@ -417,18 +418,19 @@ static void pci_read_cap(struct pci_pdev *pdev)
 	uint32_t msgctrl;
 	uint32_t len, offset, idx;
 	uint32_t table_info;
+	uint32_t pcie_devcap;
 
 	ptr = (uint8_t)pci_pdev_read_cfg(pdev->bdf, PCIR_CAP_PTR, 1U);
 
 	while ((ptr != 0U) && (ptr != 0xFFU)) {
 		cap = (uint8_t)pci_pdev_read_cfg(pdev->bdf, ptr + PCICAP_ID, 1U);
 
 		/* Ignore all other Capability IDs for now */
-		if ((cap == PCIY_MSI) || (cap == PCIY_MSIX)) {
+		if ((cap == PCIY_MSI) || (cap == PCIY_MSIX) || (cap == PCIY_PCIE)) {
```

Comment:
this line seems not needed for functionality, is it needed as the requirement of misra-c?

Reply:
Good catch. Will refine this.